swarm/src/handler/one_shot.rs

```diff
@@ -197,20 +190,17 @@ where
                 protocol: out,
                 ..
             }) => {
-                self.events_out.push(out.into());
+                self.events_out.push(Ok(out.into()));
             }
             ConnectionEvent::FullyNegotiatedOutbound(FullyNegotiatedOutbound {
                 protocol: out,
                 ..
             }) => {
                 self.dial_negotiated -= 1;
-                self.events_out.push(out.into());
+                self.events_out.push(Ok(out.into()));
             }
             ConnectionEvent::DialUpgradeError(DialUpgradeError { error, .. }) => {
-                if self.pending_error.is_none() {
-                    log::debug!("DialUpgradeError: {error}");
-                    self.keep_alive = KeepAlive::No;
```

Comment:
I chose to omit `KeepAlive::No` in the new implementation because I don't see why we shouldn't try and send the remaining requests.